lib/src/main/java/com/auth0/android/lock/LockActivity.java

```diff
@@ -372,12 +374,12 @@ public void onOAuthAuthenticationRequest(@NonNull OAuthLoginEvent event) {
             return;
         }
 
-        Map<String, Object> extraAuthParameters = null;
+        Map<String, String> extraAuthParameters = null;
         if (!TextUtils.isEmpty(event.getUsername())) {
-            extraAuthParameters = Collections.singletonMap(KEY_LOGIN_HINT, (Object) event.getUsername());
+            extraAuthParameters = Collections.singletonMap(KEY_LOGIN_HINT, event.getUsername());
         }
         Log.d(TAG, "Couldn't find an specific provider, using the default: " + WebAuthProvider.class.getSimpleName());
-        webProvider.start(this, connection, extraAuthParameters, authProviderCallback, WEB_AUTH_REQUEST_CODE);
+        webProvider.start(this, connection, extraAuthParameters, new WebCallbackWrapper(authProviderCallback));
```

Comment:
Lock works with the concept of `AuthProvider`. These are implementations from users (or our library) that know different strategies to sign in. We provide one for the web auth flow, using our `WebAuthProvider`. 

Because the `AuthProvider` interface takes an `AuthCallback` to return the results, and given that we've changed the callback type in the last SDK release, we have to wrap it in this new class `WebCallbackWrapper` (internal)